### PR TITLE
Issue 267: Add Spark Connector Examples

### LINKED ISSUE #267: Add Spark Connector Examples
Add code examples to demonstrate the capabilities of Pravega as a data stream storage system for Apache Spark

### PR DESCRIPTION
**Pull request by youminhan**
Signed-off-by: Youmin Han [Youmin.Han@dell.com](Youmin.Han@dell.com)

**Change log description**
Add code examples to demonstrate the capabilities of Pravega as a data stream storage system for Apache Spark. 

**Purpose of the change**
Fixes #267 

**What the code does**
It has various examples using the spark connector to write the event to the Pravega stream or read from it. 
Please see the `README.md` folder for details. 

**How to verify it**
See the `README.md` in `spark-connector-examples` folder for details.

### CODE REVIEW

**Review by claudiofahey — APPROVED** [review]

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Thanks for the contribution @youminhan. Please, could you change this PR to target branch `dev` instead of `master`? In this repo, the PRs are merged into `dev` until the release time, at which we merge `dev` into `master`. Please, see https://github.com/pravega/pravega-samples/wiki.

**Comment by youminhan** [comment]
> Thanks for the contribution @youminhan. Please, could you change this PR to target branch `dev` instead of `master`? In this repo, the PRs are merged into `dev` until the release time, at which we merge `dev` into `master`. Please, see https://github.com/pravega/pravega-samples/wiki.

Hey @RaulGracia , thank you for letting me know the policy for PRs. I just modified the target branch to `dev`. Let me know if there's anything I need to make changes. Thanks.

File: spark-connector-examples/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+it stores all data in memory and quickly runs out of memory.
+Using the procedure below, all data will be stored in a small HDFS cluster in Docker.
+
+In the command below, replace x.x.x.x with the IP address of a local network interface such as eth0.
+
+```
+cd
+git clone https://github.com/pravega/pravega
+cd pravega
+git checkout r0.7
```
Comment (RaulGracia) [inline spark-connector-examples/README.md:38]: Should this be `r0.8`?

File: spark-connector-examples/README.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+```
+mkdir -p ~/spark
+cd ~/spark
+tar -xzvf ~/Downloads/spark-2.4.6-bin-hadoop2.7.tgz
+ln -s spark-2.4.6-bin-hadoop2.7 current
+export PATH="$HOME/spark/current/bin:$PATH"
+```
+
+By default, the script `run_spark_ap.sh` will use an in-process Spark mini-cluster
+that is started with the Spark job ("--master local[2]").
```
Comment (RaulGracia) [inline spark-connector-examples/README.md:62]: Nit: maybe is better to use `this format` for --master local[2].

File: spark-connector-examples/README.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+### Run Examples
+
+#### Run a PySpark batch job that reads events from the file *sample_data.json* and writes to a Pravega stream
+
+```
+./run_pyspark_app.sh src/main/python/batch_file_to_pravega.py
+```
+
+The file *sample_data.json" has the following conents:
```
Comment (RaulGracia) [inline spark-connector-examples/README.md:84]: *sample_data.json" has inconsistent format.

File: spark-connector-examples/build.gradle [hunk 4]
```diff
@@ -0,0 +1,35 @@
+apply plugin: "java"
```
Comment (RaulGracia) [inline spark-connector-examples/build.gradle:1]: None of these files contains the Pravega license header. Please add them, according to the format of this repo: https://github.com/pravega/pravega

File: spark-connector-examples/run_pyspark_app.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (RaulGracia) [inline spark-connector-examples/run_pyspark_app.sh:1]: Same for script files regarding license header.

File: spark-connector-examples/src/main/python/batch_file_to_pravega.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+from pyspark.sql import SparkSession
```
Comment (RaulGracia) [inline spark-connector-examples/src/main/python/batch_file_to_pravega.py:1]: Missing license header.

File: spark-connector-examples/src/main/scala/io/pravega/example/spark/StreamPravegaToConsole.scala [hunk 7]
```diff
@@ -0,0 +1,34 @@
+package io.pravega.example.spark
```
Comment (RaulGracia) [inline spark-connector-examples/src/main/scala/io/pravega/example/spark/StreamPravegaToConsole.scala:1]: Missing license header.

**Review by RaulGracia — CHANGES_REQUESTED** [review]

**Comment by RaulGracia** [comment]
@youminhan @claudiofahey in general it LGTM, but there are mainly small doc improvements and missing license headers that require fixing.
Also, please add a section in the main README of the project for the Spark connector samples. Please, follow the structure of the other sections, in which we briefly describe the purpose of the folder and a table with the available samples and links.

File: spark-connector-examples/README.md (hunk 1 above)
Comment (youminhan) [reply spark-connector-examples/README.md]: Yep. Fixed it.

**Comment by youminhan** [comment]
@RaulGracia Thank you for taking the time to review our spark connector examples. I have updated the README and license header based on your suggestions. Please let me know if there's anything we need to change. Thanks.

**Review by RaulGracia — APPROVED** [review]
LGTM

**Pull request merged**